### ISSUE TITLE
What is the purpose of LocalConv?

### ISSUE THREAD

**Issue by zerortj** [JDAI-CV/CoTNet#11]
Why do you use the local_conv function to make Values-matrix instead of just 1*1 convolution like the paper said?

**Comment by xingshulicc** [comment]
1. The paper adopted SK block for fusion of K1 and K2, but in the code of CoTNet, you used SE.
2. I confused about the application of attention method for fusion. In fact, SK block is used to select kernel sizes, but CoTNet layer, there is no kernel difference. Maybe the difference is local and global?? So you hope the model to select to use local or global??
Can you explain these questions? Thank you

**Comment by YehLi** [comment]
> Why do you use the local_conv function to make Values-matrix instead of just 1*1 convolution like the paper said?

LocalConvolution is used for aggregating all values within each k × k grid with the learnt local attention matrix in equation 3. Section 3.4 discusses the connections between self-attention and dynamic region-aware convolution.

**Comment by YehLi** [comment]
> 1. The paper adopted SK block for fusion of K1 and K2, but in the code of CoTNet, you used SE.
> 2. I confused about the application of attention method for fusion. In fact, SK block is used to select kernel sizes, but CoTNet layer, there is no kernel difference. Maybe the difference is local and global?? So you hope the model to select to use local or global??
>    Can you explain these questions? Thank you

1. In CoT block, we use sk to fuse K1 and K2.
        x = torch.cat([x, k], dim=2)   # concatenate K1 and K2
        x_gap = x.sum(dim=2)
        x_gap = x_gap.mean((2, 3), keepdim=True)
        x_attn = self.se(x_gap)
        x_attn = x_attn.view(B, C, self.radix)
        x_attn = F.softmax(x_attn, dim=2)   # calculate the fusion scores for K1 and K2
        out = (x * x_attn.reshape((B, C, self.radix, 1, 1))).sum(dim=2)  # fuse K1 and K2

2.  We use SK to select static context (K1) and dynamic context (K2).

**Comment by Miracle-zj** [comment]
> > 1. The paper adopted SK block for fusion of K1 and K2, but in the code of CoTNet, you used SE.
> > 2. I confused about the application of attention method for fusion. In fact, SK block is used to select kernel sizes, but CoTNet layer, there is no kernel difference. Maybe the difference is local and global?? So you hope the model to select to use local or global??
> >    Can you explain these questions? Thank you
> 
> 1. In CoT block, we use sk to fuse K1 and K2.
>    x = torch.cat([x, k], dim=2)   # concatenate K1 and K2
>    x_gap = x.sum(dim=2)
>    x_gap = x_gap.mean((2, 3), keepdim=True)
>    x_attn = self.se(x_gap)
>    x_attn = x_attn.view(B, C, self.radix)
>    x_attn = F.softmax(x_attn, dim=2)   # calculate the fusion scores for K1 and K2
>    out = (x * x_attn.reshape((B, C, self.radix, 1, 1))).sum(dim=2)  # fuse K1 and K2
> 2. We use SK to select static context (K1) and dynamic context (K2).

I have a question here, is the SK in this sentence a typo, it should be SE. Thanks.
